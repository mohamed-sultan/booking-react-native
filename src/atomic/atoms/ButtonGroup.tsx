import React from 'react';
import {View, StyleSheet} from 'react-native';
import Button from './Button';
import  { useThemeColors } from '@store/useTheme';

const ButtonGroup = ({onPrevious, onContinue, disabled}: any) => {
  const colors = useThemeColors();
  const styles = getStyles(colors);
  return (
    <View style={styles.container}>
      <Button
        title="Previous"
        onPress={onPrevious}
        style={styles.secondaryButton}
        textStyle={styles.secondaryButtonText}
      />
      <Button
        disabled={disabled}
        title="Continue"
        onPress={onContinue}
        style={styles.primaryButton}
      />
    </View>
  );
};
export default React.memo(ButtonGroup);

const getStyles = (colors: any) => StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 32,
    gap: 15,
  },
  primaryButton: {
    flex: 0.5,
  },
  secondaryButton: {
    flex: 0.5,
    backgroundColor: 'white',
    borderWidth: 1,
    borderColor: colors.PrimaryBackground,
  },
  secondaryButtonText: {
    color: colors.PrimaryBackground,
  },
});

