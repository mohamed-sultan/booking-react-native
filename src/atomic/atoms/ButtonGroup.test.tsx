import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ButtonGroup from './ButtonGroup';

// Mock the useThemeColors hook
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    PrimaryBackground: '#6200ee',
  }),
}));

describe('ButtonGroup', () => {
  it('renders correctly', () => {
    const { toJSON, getByText } = render(
      <ButtonGroup onPrevious={jest.fn()} onContinue={jest.fn()} disabled={false} />
    );

    expect(getByText('Previous')).toBeTruthy();
    expect(getByText('Continue')).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when Continue button is disabled', () => {
    const { toJSON } = render(
      <ButtonGroup onPrevious={jest.fn()} onContinue={jest.fn()} disabled={true} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPrevious when Previous button is pressed', () => {
    const onPreviousMock = jest.fn();
    const { getByText } = render(
      <ButtonGroup onPrevious={onPreviousMock} onContinue={jest.fn()} disabled={false} />
    );

    fireEvent.press(getByText('Previous'));
    expect(onPreviousMock).toHaveBeenCalledTimes(1);
  });

  it('calls onContinue when Continue button is pressed', () => {
    const onContinueMock = jest.fn();
    const { getByText } = render(
      <ButtonGroup onPrevious={jest.fn()} onContinue={onContinueMock} disabled={false} />
    );

    fireEvent.press(getByText('Continue'));
    expect(onContinueMock).toHaveBeenCalledTimes(1);
  });
});
