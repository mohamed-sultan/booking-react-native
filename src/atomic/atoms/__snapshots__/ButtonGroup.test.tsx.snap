// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 15,
      "justifyContent": "space-between",
      "marginTop": 32,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#6200ee",
        "borderRadius": 8,
        "borderWidth": 1,
        "flex": 0.5,
        "marginTop": 32,
        "opacity": 1,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          },
          {
            "color": "#6200ee",
          },
        ]
      }
    >
      Previous
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "borderRadius": 8,
        "flex": 0.5,
        "marginTop": 32,
        "opacity": 1,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          },
          undefined,
        ]
      }
    >
      Continue
    </Text>
  </View>
</View>
`;

exports[`ButtonGroup renders correctly when Continue button is disabled 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 15,
      "justifyContent": "space-between",
      "marginTop": 32,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#6200ee",
        "borderRadius": 8,
        "borderWidth": 1,
        "flex": 0.5,
        "marginTop": 32,
        "opacity": 1,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          },
          {
            "color": "#6200ee",
          },
        ]
      }
    >
      Previous
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "borderRadius": 8,
        "flex": 0.5,
        "marginTop": 32,
        "opacity": 0.5,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          },
          undefined,
        ]
      }
    >
      Continue
    </Text>
  </View>
</View>
`;
